plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
I believe I found the culprit. This change: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880/commits/00e9aec21df0f6a7b373ea1f082cae50b7311008#diff-7829288eff77ee655857fff4a023693fL158

What happens here is that the error message is set to null when the paint listener is executed. However the paint listener is never called on GTK2. This is the code that decides based on the GTK version:
[Widget.java lines 1940-1951](https://github.com/eclipse/eclipse.platform.swt/blob/8be1ba1520291fd294aff6542831ec2b8d9fb0c6/bundles/org.eclipse.swt/Eclipse%20SWT/gtk/org/eclipse/swt/widgets/Widget.java#L1936)
```
if (OS.GTK3) {
  if (OS.GTK_VERSION >= OS.VERSION (3, 9, 0) && OS.GTK_IS_CONTAINER (handle)) {
    return gtk_draw (handle, arg0);
  }
} else {
  GdkEventExpose gdkEvent = new GdkEventExpose ();
  OS.memmove (gdkEvent, arg0, GdkEventExpose.sizeof);
  long /*int*/ paintWindow = paintWindow ();
  long /*int*/ window = gdkEvent.window;
  if (window != paintWindow) return 0;
    return (state & OBSCURED) != 0 ? 1 : 0;
}
```

So when the referenced commit removed the `setErrorMessage(null)` call GTK3 did not remove the error message anymore in the paint listener.